src/components/ServiceRequestForm.jsx

```diff
@@ -0,0 +1,127 @@
+import React, { useState } from "react";
+import { Formik, Form, Field } from "formik";
+import Categories from "./services/categories.json";
+import * as Yup from "yup";
+import ErrorMsg from "./errormsg";
+import FormContainer from './FormContainer';
+
+const getSubCategories = (categories, categoryId) => {
+	var category = categories.find(category => category.id === categoryId);
+	return category ? category.types : [];
+};
+
+const ServiceRequestForm = props => {
+
+	const [categories] = useState(Categories);
+	const [subCategories, setSubCategories] = useState([]);
+
+	const handleServiceRequestChange = (changeEvent, setFieldValue) => {
+		const { value } = changeEvent.currentTarget;
+		const subCategories = getSubCategories(categories, parseInt(value));
+		setFieldValue(changeEvent.currentTarget.name, changeEvent.currentTarget.value)
+		setFieldValue('subRequestType', '')
+		setSubCategories(subCategories);
+	};
+
+
+	return (
+
+		<React.Fragment>
+			<FormContainer title="How Can We Help?">
+				<Formik
+					initialValues={{
+						requestType: '',
+						subRequestType: '',
+					}}
+					validationSchema={Yup.object().shape({
+						requestType: Yup.string().required('Request Category is required'),
+						subRequestType: Yup.string().required('Sub Category is required'),
+					})}
+
+					onSubmit={(values, { setSubmitting }) => {
+
+						alert(JSON.stringify(values, null, 2));
+						setSubmitting(false);
+
+					}}
+				>
+					{
+						(props) => {
+							const { values, isSubmitting, errors, touched } = props;
+
+							return (
+
+								<Form >
+									<label htmlFor="requestType"
+										className={
+											errors.requestType && touched.requestType ? "input-feedback" : "text-label"}
+
+									>Request Category</label>
+									<br />
+									<Field
+										component="select"
+										name="requestType"
+										onChange={e => { handleServiceRequestChange(e, props.setFieldValue) }}
```

Comment:
```js
onChange={handleServiceRequestChange}
```

not sure if it came through in the last but if you call you don't want to call an anonymous function here:

```js
e => { handleServiceRequestChange(e, props.setFieldValue) }
```
becasue it will declare a new function everytime it changes

we want to use the one handleServiceRequestChange to handle the change. If you are having difficulties getting this to work, please let me know and I will help you.

Reply:
changes onchange event